### DISCUSSION TITLE
How to directly write data via Bluetooth?

### DISCUSSION THREAD

**Discussion by van9petryk** [capacitor-community/bluetooth-le#584] [Q&A] [answered]
I used old BluetoothSerial cordova plugin
https://www.npmjs.com/package/cordova-plugin-bluetooth-serial
and wrote directly text to bluetooth device.

The sample of code was
```
this.bluetoothSerial.connect(MAC).subscribe(r => {
    this.bluetoothSerial.write(template)
})
```

I decided to switch to capacitor plugins, since it is recommended by ionic, but don't understand how to do same with bluetooth-le capacitor plugin.

**Comment by van9petryk** [comment]
and where to take UID of services? I see it hardcoded in the example.
```
const HEART_RATE_SERVICE = '0000180d-0000-1000-8000-00805f9b34fb';
const HEART_RATE_MEASUREMENT_CHARACTERISTIC = '00002a37-0000-1000-8000-00805f9b34fb';
const BODY_SENSOR_LOCATION_CHARACTERISTIC = '00002a38-0000-1000-8000-00805f9b34fb';
const BATTERY_SERVICE = numberToUUID(0x180f);
const BATTERY_CHARACTERISTIC = numberToUUID(0x2a19);
const POLAR_PMD_SERVICE = 'fb005c80-02e7-f387-1cad-8acd2d8df0c8';
const POLAR_PMD_CONTROL_POINT = 'fb005c81-02e7-f387-1cad-8acd2d8df0c8';
```

**Comment by pwespi** [comment, marked as answer]
This plugin only supports Bluetooth **Low Energy**, not Bluetooth **classic / serial**.

For Bluetooth classic there is e.g. this plugin: https://github.com/matfire/ionic_bluetooth_classic (I have never used it).

For the UUIDS, they are part of Bluetooth Low Energy. Some of them are assigned by the specification, see https://www.bluetooth.com/specifications/assigned-numbers/. Others are just chosen values by the peripheral Bluetooth device.
